common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -381,17 +381,19 @@ static void setMemory(Object o, long offset, long bytes, byte value) {
     }
 
     static boolean equals(byte[] bytes1, int startPos1, byte[] bytes2, int startPos2, int length) {
-        if (length == 0) {
-            return true;
-        }
         final long baseOffset1 = BYTE_ARRAY_BASE_OFFSET + startPos1;
         final long baseOffset2 = BYTE_ARRAY_BASE_OFFSET + startPos2;
         final int remainingBytes = length & 7;
-        final long end = baseOffset1 + remainingBytes;
-        for (long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length; i >= end; i -= 8, j -= 8) {
-            if (UNSAFE.getLong(bytes1, i) != UNSAFE.getLong(bytes2, j)) {
-                return false;
-            }
+        if (length > 7) {
+            final long end = baseOffset1 + remainingBytes;
+            long i = baseOffset1 - 8 + length, j = baseOffset2 - 8 + length;
+            do {
+                if (UNSAFE.getLong(bytes1, i) != UNSAFE.getLong(bytes2, j)) {
```

Comment:
@carl-mastrangelo - Here are the results of `PlatformDependentBenchmark`. I switched the measurement to average time and added `1` for a size.

We gain on the smaller inputs, but start to perform worse around input size of `100`. However the gain may be negligible if there are larger headers involved ... let me revert this.

```
With this PR
Benchmark                                    (size)  Mode  Cnt     Score     Error  Units
PlatformDependentBenchmark.unsafeBytesEqual       1  avgt    5     4.453 ±   0.081  ns/op
PlatformDependentBenchmark.unsafeBytesEqual       7  avgt    5     4.756 ±   0.139  ns/op
PlatformDependentBenchmark.unsafeBytesEqual      10  avgt    5     5.493 ±   0.223  ns/op
PlatformDependentBenchmark.unsafeBytesEqual      50  avgt    5     9.630 ±   0.262  ns/op
PlatformDependentBenchmark.unsafeBytesEqual     100  avgt    5    13.559 ±   0.306  ns/op
PlatformDependentBenchmark.unsafeBytesEqual    1000  avgt    5   107.538 ±   1.410  ns/op
PlatformDependentBenchmark.unsafeBytesEqual   10000  avgt    5   995.796 ±  16.488  ns/op
PlatformDependentBenchmark.unsafeBytesEqual  100000  avgt    5  9942.613 ± 350.996  ns/op

Before This PR
Benchmark                                    (size)  Mode  Cnt     Score     Error  Units
PlatformDependentBenchmark.unsafeBytesEqual       1  avgt    5     4.906 ±   0.404  ns/op
PlatformDependentBenchmark.unsafeBytesEqual       7  avgt    5     5.469 ±   0.099  ns/op
PlatformDependentBenchmark.unsafeBytesEqual      10  avgt    5     6.051 ±   0.098  ns/op
PlatformDependentBenchmark.unsafeBytesEqual      50  avgt    5     9.634 ±   0.235  ns/op
PlatformDependentBenchmark.unsafeBytesEqual     100  avgt    5    12.056 ±   0.455  ns/op
PlatformDependentBenchmark.unsafeBytesEqual    1000  avgt    5    73.085 ±   1.784  ns/op
PlatformDependentBenchmark.unsafeBytesEqual   10000  avgt    5   649.883 ±  23.863  ns/op
PlatformDependentBenchmark.unsafeBytesEqual  100000  avgt    5  7254.138 ± 475.926  ns/op
```